File: villains-service/src/main/java/me/escoffier/villains/VillainResource.java
package me.escoffier.villains;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import java.util.List;

@Path("/")
public class VillainResource {

    @GET
    public Villain get() {
        return Villain.getRandomVillain();
    }
}

File: villains-service/src/main/kotlin/me/escoffier/villains/VillainResource.kt
package me.escoffier.villains

import io.smallrye.mutiny.Multi
import io.smallrye.mutiny.coroutines.awaitSuspending
import javax.ws.rs.GET
import javax.ws.rs.Path

@Path("/")
class VillainResource {

    @GET
    @Path("/all")
    fun getAll() : Multi<Villain> {
        return Villain.streamAll();
    }

    @GET
    suspend fun getRandomVillain() : Villain {
        return Villain.getRandomVillain().awaitSuspending()
    }

}